test/src/test/java/org/corfudb/runtime/view/StreamViewTest.java

```diff
@@ -464,4 +471,75 @@ public void testPreviousWithStreamCheckpoint() throws Exception {
         assertThatThrownBy(() -> sv.previous()).isInstanceOf(TrimmedException.class);
         assertThat(sv.getCurrentGlobalPosition()).isEqualTo(baseVersion.getSequence());
     }
+
+    /**
+     * Ensure that transaction stream semantics are correct with respect to different
+     * combinations of trim points and seek positions.
+     *
+     * @throws InterruptedException
+     */
+    @Test
+    public void txLogTrim() throws InterruptedException {
+        final CorfuRuntime.CorfuRuntimeParameters params = CorfuRuntime.CorfuRuntimeParameters
+                .builder()
+                .backpointersDisabled(true)
+                .build();
+        final CorfuRuntime localRuntime = CorfuRuntime.fromParameters(params)
+                .setTransactionLogging(true)
+                .parseConfigurationString(getDefaultConfigurationString())
+                .connect();
+        final CorfuTable<String, String>
+                instance1 = localRuntime.getObjectsView().build()
+                .setTypeToken(new TypeToken<CorfuTable<String, String>>() {})
+                .setStreamName("txTestMap")
+                .open();
+
+        final Duration sleepDuration = Duration.ofSeconds(5);
+
+        // Populate the Transaction Stream up to NUM_ITERATIONS_LOW entries.
+        IntStream.range(0, PARAMETERS.NUM_ITERATIONS_LOW).forEach(idx -> {
+            localRuntime.getObjectsView().TXBegin();
+            instance1.put(String.valueOf(idx), String.valueOf(idx));
+            localRuntime.getObjectsView().TXEnd();
+        });
+
+        // Force a prefix trim.
+        final long epoch = 0L;
+        localRuntime.getAddressSpaceView().prefixTrim(
+                new Token(epoch, PARAMETERS.NUM_ITERATIONS_LOW/2/2));
+
+        // Wait until log is trimmed.
+        Thread.sleep(sleepDuration.toMillis());
```

Comment:
Putting sleeps in the test is a bad idea and makes these tests flaky. Travis environments are slow and might behave differently. Can you query the trimMark and busy-wait on that?

Reply:
I mostly copied the test from somewhere else and made the necessary changes.

But yea -- will do.

Reply:
Done.